src/Sentry/buildTransitive/Sentry.targets

```diff
@@ -120,53 +125,88 @@
-    <Message Importance="High"
-      Text="Preparing to upload debug symbols to Sentry for $(MSBuildProjectName) ($(Configuration)/$(TargetFramework))" />
-
-    <ItemGroup>
-      <SentryCLIUploadSymbolType Include="pdb" />
-      <SentryCLIUploadSymbolType Include="portablepdb" />
-      <SentryCLIUploadSymbolType Include="dsym" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'" />
-      <SentryCLIUploadSymbolType Include="dsym" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'" />
-    </ItemGroup>
-
-    <Exec Command="$(SentryCLIDebugFilesUploadCommand) @(SentryCLIUploadSymbolType -> '-t %(Identity)', ' ') $(OutputPath)" IgnoreExitCode="true" ContinueOnError="WarnAndContinue">
-      <Output TaskParameter="ExitCode" PropertyName="_SentryCLIExitCode" />
-    </Exec>
+  <!-- Native AOT publishing
+       While '_IsPublishing' looks a bit "private", it's also OK if it suddenly stops working and this step runs anyway.
+       See https://github.com/dotnet/sdk/issues/26324#issuecomment-1169236993 -->
+  <PropertyGroup Condition="'$(PublishDir)' != '' and '$(PublishAot)' == 'true' and '$(_IsPublishing)' == 'true'">
+    <SentryCLIUploadNativeAOT>true</SentryCLIUploadNativeAOT>
+    <SentryCLIUploadDirectory>$(PublishDir)</SentryCLIUploadDirectory>
+    <!-- We must run after "CopyNativeBinary" (not "Publish"),
+         because that is the one that actually copies native AOT symbols to the publish directory. -->
+    <SentryCLIUploadAfterTargets>CopyNativeBinary</SentryCLIUploadAfterTargets>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(SentryCLIUploadNativeAOT)' == 'true'">
+    <SentryCLIUploadSymbolType Include="pdb" />
+    <SentryCLIUploadSymbolType Include="dsym" />
+    <SentryCLIUploadSymbolType Include="elf" />
+  </ItemGroup>
+
+  <!-- "Standard" managed build -->
+  <PropertyGroup Condition="'$(SentryCLIUploadNativeAOT)' != 'true'">
+    <SentryCLIUploadNativeAOT>false</SentryCLIUploadNativeAOT>
+    <SentryCLIUploadDirectory>$(OutputPath)</SentryCLIUploadDirectory>
+    <SentryCLIUploadAfterTargets>Build</SentryCLIUploadAfterTargets>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(SentryCLIUploadNativeAOT)' != 'true'">
+    <SentryCLIUploadSymbolType Include="pdb" />
+    <SentryCLIUploadSymbolType Include="portablepdb" />
+    <SentryCLIUploadSymbolType Include="dsym" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'" />
+    <SentryCLIUploadSymbolType Include="dsym" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'" />
+  </ItemGroup>
 
+  <!-- Upload symbols to Sentry after the build. -->
+  <Target Name="UploadDebugInfoToSentry" AfterTargets="$(SentryCLIUploadAfterTargets)" DependsOnTargets="PrepareSentryCLI"
+    Condition="'$(SentryCLI)' != '' and ('$(SentryUploadSymbols)' == 'true' or '$(SentryUploadSources)' == 'true')">
+
+    <!-- if (UploadSymbols && UploadSources) { -->
```

Comment:
Thanks for those!

Reply:
yeah they're nice although at one point I've found myself just copying a piece of a code from one if branch to the other, forgetting to actually update the relevant condition 😅